src/frequenz/sdk/_internal/_channels.py

```diff
@@ -24,3 +25,26 @@ def new_receiver(self, maxsize: int = 50) -> Receiver[T]:
         Returns:
             A receiver instance.
         """
+
+
+class LatestValueCache(typing.Generic[T]):
+    """A cache that stores the latest value in a receiver."""
+
+    def __init__(self, receiver: Receiver[T]) -> None:
+        """Create a new cache.
+
+        Args:
+            receiver: The receiver to cache.
+        """
+        self._receiver = receiver
+        self._latest_value: T | None = None
```

Comment:
Shouldn't this use a sentinel? `None` could be a valid `T` (if `T` is, for example, `float | None`, like in `Sample.value`).

Reply:
oh true 😭 

will fix on Monday.

Reply:
https://github.com/frequenz-floss/frequenz-sdk-python/pull/846